File: web/apps/web-docs-hub/docs/examples/hello_world.js
/**
 * Igris Inertial — Node.js hello world
 * Requires: npm install @igris-inertial/sdk
 *
 * Usage:
 *   export IGRIS_API_KEY=igris_...
 *   node hello_world.js
 */

import { IgrisClient } from '@igris-inertial/sdk';

const apiKey = process.env.IGRIS_API_KEY;
if (!apiKey) {
  console.error('Set IGRIS_API_KEY first: export IGRIS_API_KEY=igris_...');
  process.exit(1);
}

const client = new IgrisClient({
  baseUrl: 'https://overture.igrisinertial.com',
  apiKey,
});

const response = await client.infer({
  model: 'gpt-4',
  messages: [{ role: 'user', content: 'Hello! Reply in one sentence.' }],
  max_tokens: 64,
});

console.log(response.choices[0].message.content);
console.log(`Provider : ${response.metadata?.provider ?? '—'}`);
console.log(`Latency  : ${response.metadata?.latency_ms ?? '—'} ms`);
console.log(`Cost     : $${response.metadata?.cost_usd?.toFixed(6) ?? '—'}`);

File: web/apps/web-docs-hub/docs/examples/hello_world.py
"""
Igris Inertial — Python hello world
Requires: pip install igris-inertial

Usage:
  export IGRIS_API_KEY=igris_...
  python hello_world.py
"""

import os
from igris import IgrisClient, InferRequest, Message

api_key = os.environ.get("IGRIS_API_KEY")
if not api_key:
    raise SystemExit("Set IGRIS_API_KEY first: export IGRIS_API_KEY=igris_...")

client = IgrisClient(
    base_url="https://overture.igrisinertial.com",
    api_key=api_key,
)

response = client.infer(InferRequest(
    model="gpt-4",
    messages=[Message(role="user", content="Hello! Reply in one sentence.")],
    max_tokens=64,
))

print(response.choices[0].message.content)
print(f"Provider : {response.metadata.provider if response.metadata else '—'}")
print(f"Latency  : {response.metadata.latency_ms if response.metadata else '—'} ms")
print(f"Cost     : ${response.metadata.cost_usd:.6f}" if response.metadata else "Cost: —")
